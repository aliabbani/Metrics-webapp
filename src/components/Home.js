/* eslint-disable react/prop-types */
import React from 'react';
import { AiOutlineRightCircle } from 'react-icons/ai';
import { NavLink } from 'react-router-dom';
import '../stylesheet/home.css';
import { BiWorld } from 'react-icons/bi';

const Home = (props) => {
  const { datas, totals } = props;

  return (
    <div className="main-container">
      <div className="header">
        <div className="header-left">
          <BiWorld className="header-icon" />
        </div>
        <div className="header-right">
          <h1 className="header-h1">WorldWide</h1>
          {totals.map((total) => (
            <h4 key={total.date} className="header-h4">
              {total.today_confirmed}
              {' '}
              cases
            </h4>
          ))}
        </div>
      </div>

      <h5 className="home-title">STATS BY COUNTRY</h5>

      <div className="container-grid">
        {datas.map((data) => {
          const test = `/detail/${data.id}`;
          return (
            <div key={data.id} className={datas.indexOf(data) % 2 === 0 ? 'inside-flex' : 'inside-flex-bolder'}>
              <NavLink className="nav-icon" to={{ pathname: test, data }} exact>
                <AiOutlineRightCircle className="inside-up" />
              </NavLink>
              <div className="inside-down">
                <h3 className="inside-h3">{data.name}</h3>
                <div className="confirmed">
                  <h4 className="inside-h4">{data.today_confirmed}</h4>
                </div>
              </div>
            </div>
          );
        })}
      </div>
    </div>
  );
};

export default Home;

// console.log('here is my HOME data', datas);
// let total = 0;
// for (let i = 0; i < datas.length; i = +1) {
//   total += datas[i].today_confirmed;
// }
// console.log(total);
// const array1 = [1, 2, 3, 4];
// const reducer = (previousValue, currentValue) => previousValue + currentValue;
// const resul = datas.today_confirmed.reduce(reducer);

// <div className="header-right">
//   <h1 className="header-h1">WorldWide</h1>
//   {totals.map((total) => (
//     <h4 key={total.date} className="header-h4">
//       {total.today_confirmed}
//       {' '}
//       cases
//     </h4>
//   ))}
// </div>;
