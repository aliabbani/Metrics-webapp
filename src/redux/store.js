import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import dataReducer from './reducers/dataReducer';

const reducer = combineReducers({
  dataReducer,
});
const store = createStore(
  reducer,
  applyMiddleware(logger, thunk),
);

export default store;
