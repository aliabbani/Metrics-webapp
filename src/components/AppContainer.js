import React, { useEffect } from 'react';
import { Route, Switch } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import Navbar from './Navbar';
import NotMatch from '../pages/NotMatch';
import Home from './Home';
import Details from './Details';
import { fetchPostsRequestData } from '../redux/reducers/dataReducer';

const AppContainer = () => {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchPostsRequestData());
  }, []);

  const datas = useSelector((state) => state.dataReducer.data);
  console.log('here is my DATAS', datas);

  return (
    <div>
      <Navbar />
      <Switch>
        <Route exact path="/">
          <Home datas={datas} />
        </Route>
        <Route path="/details">
          <Details />
        </Route>
        <Route path="*">
          <NotMatch />
        </Route>
      </Switch>
    </div>
  );
};

export default AppContainer;
