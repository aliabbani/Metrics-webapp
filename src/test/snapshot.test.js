import React from 'react';
import renderer from 'react-test-renderer';
import { NavLink, BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import { IoIosArrowBack } from 'react-icons/io';
import AppContainer from '../components/AppContainer';
import Home from '../components/Home';
// import Details from '../components/Details';
import store from '../redux/store';
// import Navbar from '../components/Navbar';

describe('Component test snapshot', () => {
  it('App renders correctly', () => {
    const tree = renderer
      .create(
        <Router>
          <Provider store={store}>
            <AppContainer />
          </Provider>
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Router renders correctly', () => {
    const tree = renderer
      .create(
        <Router>
          <NavLink to="/"><IoIosArrowBack /></NavLink>
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Home page renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Home />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
