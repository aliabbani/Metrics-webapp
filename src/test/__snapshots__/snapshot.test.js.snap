// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component test snapshot App renders correctly 1`] = `
<div
  className="app-container"
>
  <nav
    className="navBar"
  >
    <a
      aria-current="page"
      className="left-nav active-link"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M217.9 256L345 129c9.4-9.4 9.4-24.6 0-33.9-9.4-9.4-24.6-9.3-34 0L167 239c-9.1 9.1-9.3 23.7-.7 33.1L310.9 417c4.7 4.7 10.9 7 17 7s12.3-2.3 17-7c9.4-9.4 9.4-24.6 0-33.9L217.9 256z"
        />
      </svg>
    </a>
    <h4
      className="middle-nav"
    >
      COVID-19 DATA
    </h4>
    <div
      className="right-nav"
    >
      <svg
        className="mic"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M5 3a3 3 0 0 1 6 0v5a3 3 0 0 1-6 0V3z"
        />
        <path
          d="M3.5 6.5A.5.5 0 0 1 4 7v1a4 4 0 0 0 8 0V7a.5.5 0 0 1 1 0v1a5 5 0 0 1-4.5 4.975V15h3a.5.5 0 0 1 0 1h-7a.5.5 0 0 1 0-1h3v-2.025A5 5 0 0 1 3 8V7a.5.5 0 0 1 .5-.5z"
        />
      </svg>
      <svg
        className="gear"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9.405 1.05c-.413-1.4-2.397-1.4-2.81 0l-.1.34a1.464 1.464 0 0 1-2.105.872l-.31-.17c-1.283-.698-2.686.705-1.987 1.987l.169.311c.446.82.023 1.841-.872 2.105l-.34.1c-1.4.413-1.4 2.397 0 2.81l.34.1a1.464 1.464 0 0 1 .872 2.105l-.17.31c-.698 1.283.705 2.686 1.987 1.987l.311-.169a1.464 1.464 0 0 1 2.105.872l.1.34c.413 1.4 2.397 1.4 2.81 0l.1-.34a1.464 1.464 0 0 1 2.105-.872l.31.17c1.283.698 2.686-.705 1.987-1.987l-.169-.311a1.464 1.464 0 0 1 .872-2.105l.34-.1c1.4-.413 1.4-2.397 0-2.81l-.34-.1a1.464 1.464 0 0 1-.872-2.105l.17-.31c.698-1.283-.705-2.686-1.987-1.987l-.311.169a1.464 1.464 0 0 1-2.105-.872l-.1-.34zM8 10.93a2.929 2.929 0 1 1 0-5.86 2.929 2.929 0 0 1 0 5.858z"
        />
      </svg>
    </div>
  </nav>
  <div
    className="main-container"
  >
    <div
      className="header"
    >
      <div
        className="header-left"
      >
        <svg
          className="header-icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 2C6.486 2 2 6.486 2 12s4.486 10 10 10 10-4.486 10-10S17.514 2 12 2zM4 12c0-.899.156-1.762.431-2.569L6 11l2 2v2l2 2 1 1v1.931C7.061 19.436 4 16.072 4 12zm14.33 4.873C17.677 16.347 16.687 16 16 16v-1a2 2 0 0 0-2-2h-4v-3a2 2 0 0 0 2-2V7h1a2 2 0 0 0 2-2v-.411C17.928 5.778 20 8.65 20 12a7.947 7.947 0 0 1-1.67 4.873z"
          />
        </svg>
      </div>
      <div
        className="header-right"
      >
        <h1
          className="header-h1"
        >
          WorldWide
        </h1>
      </div>
    </div>
    <h5
      className="home-title"
    >
      STATS BY COUNTRY
    </h5>
    <div
      className="container-grid"
    />
  </div>
</div>
`;

exports[`Component test snapshot Home page renders correctly 1`] = `
<div
  className="main-container"
>
  <div
    className="header"
  >
    <div
      className="header-left"
    >
      <svg
        className="header-icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 2C6.486 2 2 6.486 2 12s4.486 10 10 10 10-4.486 10-10S17.514 2 12 2zM4 12c0-.899.156-1.762.431-2.569L6 11l2 2v2l2 2 1 1v1.931C7.061 19.436 4 16.072 4 12zm14.33 4.873C17.677 16.347 16.687 16 16 16v-1a2 2 0 0 0-2-2h-4v-3a2 2 0 0 0 2-2V7h1a2 2 0 0 0 2-2v-.411C17.928 5.778 20 8.65 20 12a7.947 7.947 0 0 1-1.67 4.873z"
        />
      </svg>
    </div>
    <div
      className="header-right"
    >
      <h1
        className="header-h1"
      >
        WorldWide
      </h1>
    </div>
  </div>
  <h5
    className="home-title"
  >
    STATS BY COUNTRY
  </h5>
  <div
    className="container-grid"
  />
</div>
`;

exports[`Component test snapshot Router renders correctly 1`] = `
<a
  aria-current="page"
  className="active"
  href="/"
  onClick={[Function]}
  style={Object {}}
>
  <svg
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 512 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M217.9 256L345 129c9.4-9.4 9.4-24.6 0-33.9-9.4-9.4-24.6-9.3-34 0L167 239c-9.1 9.1-9.3 23.7-.7 33.1L310.9 417c4.7 4.7 10.9 7 17 7s12.3-2.3 17-7c9.4-9.4 9.4-24.6 0-33.9L217.9 256z"
    />
  </svg>
</a>
`;
